source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1698,9 +1701,9 @@ must do no I/O::
         address = server.address
         newTopologyDescription.servers[address] = server
 
-        # for drivers that implement CMAP, mark the connection pool as ready after
-        # a successful check on a data bearing server.
-        if server.type in (Mongos, RSPrimary, RSSecondary, Standalone, LoadBalanced):
+        # for drivers that implement CMAP, mark the connection pool as ready after a successful check
+        if (server.type in (Mongos, RSPrimary, RSSecondary, Standalone, LoadBalanced))
+                or (server.type != Unknown and directConnection):
```

Comment:
directConnection -> "newTopologyDescription.type == Single"

Reply:
`newTopologyDescription.type == Single` only if `server.type` is `Standalone` and is covered by the spec in its state even before the PR . My understanding is that `directConnection` may be requested for any deployment (not only standalone), and in that case we must create + unpause a pool regardless of the discovered server type - this is what this PR tries to achieve.

Reply:
> newTopologyDescription.type == Single only if server.type is Standalone.

"TopologyType==single" is not the same as "server.type==Standalone". With TopologyType == single, the single server can be of _any_ type.

Reply:
> With TopologyType == single, the single server can be of any type.

It appears to me that the spec says otherwise. When I look at https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topologytype-table, I see that `updateUnknownWithStandalone` is called only if `server.type==Standalone"`. [`updateUnknownWithStandalone`](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#updateunknownwithstandalone) sets `topologyDescription.type = Single` and as far as I can see there are no other ways to have `newTopologyDescription.type == Single`.

Reply:
On the other hand, there is also this: _"[TopologyType Single](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topologytype-single). The TopologyDescription's type was initialized as Single and remains Single forever. "_

Does the Python driver sometimes sets `TopologyType Single` at the very beginning, i.e., does not start with topology type `Unknown` without user requesting `directConnection`? If yes, I am curious when this happen.